_studio/mfx_lib/encode_hw/h265/src/mfx_h265_encode_vaapi.cpp

```diff
@@ -679,18 +679,37 @@ void UpdateSlice(
                 slice.ref_pic_list1[ref].flags         = VA_PICTURE_HEVC_INVALID;
             }
         }
-/*
-        slice.luma_log2_weight_denom;
-        slice.delta_chroma_log2_weight_denom
-        slice.delta_luma_weight_l0
-        slice.luma_offset_l0
-        slice.delta_chroma_weight_l0
-        slice.chroma_offset_l0
-        slice.delta_luma_weight_l1
-        slice.luma_offset_l1
-        slice.delta_chroma_weight_l1
-        slice.chroma_offset_l1
-*/
+
+#if defined(MFX_ENABLE_HEVCE_WEIGHTED_PREDICTION)
+    if ((slice.slice_type == SLICE_P && pps.pic_fields.bits.weighted_pred_flag) ||
+        (slice.slice_type == SLICE_B && pps.pic_fields.bits.weighted_bipred_flag))
+    {
+        const mfxU16 Y = 0, Cb = 1, Cr = 2, W = 0, O = 1;
```

Comment:
O looks too much like 0. Possible to rename?

Reply:
Done